Framerate Converter (#1)
membraneframework/membrane_framerate_converter_plugin: README.md

# Membrane Framerate Converter Plugin

[![Hex.pm](https://img.shields.io/hexpm/v/membrane_framerate_converter_plugin.svg)](https://hex.pm/packages/membrane_framerate_converter_plugin)
[![API Docs](https://img.shields.io/badge/api-docs-yellow.svg?style=flat)](https://hexdocs.pm/membrane_framerate_converter_plugin)
[![CircleCI](https://circleci.com/gh/membraneframework/membrane_framerate_converter_plugin.svg?style=svg)](https://circleci.com/gh/membraneframework/membrane_framerate_converter_plugin)

> Plugin providing element for converting framerate of raw video stream.

Reviewer: I would stick to either `frame rate` or `framework` throughout the README
```suggestion
Plugin providing element for converting frame rate of raw video stream.
```

It is part of [Membrane Multimedia Framework](https://membraneframework.org).

## Installation

The package can be installed by adding `membrane_framerate_converter_plugin` to your list of dependencies in `mix.exs`:

```elixir
def deps do
  [
    {:membrane_framerate_converter_plugin, "~> 0.1.0"}
  ]
end
```

## Description

Element converts video to target constant frame rate, by dropping and duplicating frames as necessary
(input video may have constant or variable frame rate).

## Usage

Example converting h264 video from 10 to 2 fps.

```elixir
defmodule Pipeline do
  use Membrane.Pipeline
  alias Membrane.H264.FFmpeg.{Parser, Decoder, Encoder}

>   alias Membrane.File.{Sink, Source}

Reviewer: ```suggestion
  use Membrane.Pipeline
  
  alias Membrane.H264.FFmpeg.{Parser, Decoder, Encoder}
  alias Membrane.File.{Sink, Source}
```

  alias Membrane.File.{Sink, Source}

  @impl true
  def handle_init(filename) do

>     children =%{

Reviewer: ```suggestion
    children = %{
```

        file: %Source{chunk_size: 40_960, location: filename},

>         parser: %Parser{framerate: {10, 1}},

Reviewer: You can also copy that somewhere, write `mix format` and copy back here to make sure it is formatted properly
```suggestion
  def handle_init(filename) do
    children = %{
        file: %Source{chunk_size: 40_960, location: filename},
        parser: %Parser{framerate: {10, 1}},
```
